### PR TITLE
LPS-196070 Create the 'ERC' field to Object Validations Entity and use it in the putObjectDefinition logic

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPS-196070](https://liferay.atlassian.net/browse/LPS-196070)

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java [hunk 1]
```diff
@@ -109,6 +109,7 @@ public void testAddObjectValidationRule() throws Exception {
 			() -> _addObjectValidationRule(
 				ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
 				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				RandomTestUtil.randomString(),
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java:112]: Shouldn't this be `StringPool.BLANK`?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java [hunk 2]
```diff
@@ -117,7 +118,7 @@ public void testAddObjectValidationRule() throws Exception {
 			() -> _addObjectValidationRule(
 				ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
 				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
-				null, _VALID_DDM_SCRIPT));
+				RandomTestUtil.randomString(), null, _VALID_DDM_SCRIPT));
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java:121]: same here

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java [hunk 3]
```diff
@@ -154,7 +155,7 @@ public void testAddObjectValidationRule() throws Exception {
 					NAME_OUTPUT_OBJECT_FIELD_ID),
 			() -> _addObjectValidationRule(
 				ObjectValidationRuleConstants.ENGINE_TYPE_DDM, errorLabelMap,
-				nameLabelMap,
+				RandomTestUtil.randomString(), nameLabelMap,
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java:158]: same here?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleServiceTest.java [hunk 4]
```diff
@@ -264,6 +266,7 @@ private void _testUpdateObjectValidationRule(User user) throws Exception {
 
 			objectValidationRule =
 				_objectValidationRuleService.updateObjectValidationRule(
+					RandomTestUtil.randomString(),
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleServiceTest.java:269]: `StringPool.BLANK`?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java [hunk 5]
```diff
@@ -287,7 +295,8 @@ public void testUpdateObjectValidationRule() throws Exception {
 				"No ObjectValidationRule exists with the primary key %s",
 				randomId),
 			() -> _objectValidationRuleLocalService.updateObjectValidationRule(
-				randomId, false, ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
+				RandomTestUtil.randomString(), randomId, false,
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java:298]: `StringPool.BLANK`?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 6]
```diff
@@ -94,13 +96,19 @@ public ObjectValidationRule addObjectValidationRule(
...
 		ObjectValidationRule objectValidationRule =
 			objectValidationRulePersistence.create(
 				counterLocalService.increment());
 
+		if (Validator.isNull(externalReferenceCode)) {
+			externalReferenceCode = objectValidationRule.getUuid();
+		}
+
+		objectValidationRule.setExternalReferenceCode(externalReferenceCode);
+
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:111]: Empty row not needed

**Review by gabrielwas — CHANGES_REQUESTED** [review]
Check my comments, please!

You did a great job on the requested changes! Thx.

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Review by gabrielwas — APPROVED** [review]
LGTM! Great job @pedro-oliveira446 !
Thx!

**Comment by pedro-oliveira446** [comment]
Resent [here](https://github.com/brianchandotcom/liferay-portal/pull/141316)

**Pull request closed without merging**